frontend/src/wasm-communication/messages.ts

```diff
@@ -1214,115 +1249,127 @@ export function isWidgetSection(layoutRow: LayoutGroup): layoutRow is WidgetSect
 
 // Unpacking rust types to more usable type in the frontend
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
-function createWidgetLayout(widgetLayout: any[]): LayoutGroup[] {
-	return widgetLayout.map((layoutType): LayoutGroup => {
-		if (layoutType.column) {
-			const columnWidgets = hoistWidgetHolders(layoutType.column.columnWidgets);
-
-			const result: WidgetColumn = { columnWidgets };
-			return result;
+function createWidgetDiff(diffs: any[]): WidgetDiff[] {
+	const diff = diffs.map((diff): WidgetDiff => {
+		const { path, newVal } = diff;
+		if (newVal.subLayout) {
+			return { path, new: newVal.subLayout.map(createLayoutGroup) };
+		}
+		if (newVal.layoutGroup) {
+			return { path, new: createLayoutGroup(newVal.layoutGroup) };
+		}
+		if (newVal.widget) {
+			return { path, new: parseWidgetHolder(newVal.widget) };
 		}
+		throw new Error("DiffUpdate invalid");
+	});
+	return diff;
+}
 
-		if (layoutType.row) {
-			const rowWidgets = hoistWidgetHolders(layoutType.row.rowWidgets);
+// Unpacking a layout group
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function createLayoutGroup(newVal: any): LayoutGroup {
+	if (newVal.column) {
```

Comment:
Do not really like the name `newVal` here. Can we use something more descriptive here and above as well?